src/components/AuthForm.tsx

```diff
@@ -0,0 +1,146 @@
+import React, { useState, useContext } from "react"
+import { useHistory } from "react-router-dom"
+import { makeStyles } from "@material-ui/core/styles"
+import { InputAdornment, TextField } from "@material-ui/core"
+import LogoutImage from "../assets/logout-image.svg"
+import classes from "./AuthForm.module.css"
+import { AccountCircle } from "@material-ui/icons"
+import PrimaryButton from "../UI/PrimaryButton"
+import IconButton from "@material-ui/core/IconButton"
+import Visibility from "@material-ui/icons/Visibility"
+import VisibilityOff from "@material-ui/icons/VisibilityOff"
+import { AuthContext } from "../store/auth-context"
+// import {RouteComponentProps} from 'react-router-dom'
+
+const useStyles = makeStyles(() => ({
+  logoutField: {
+    width: "30ch",
+  },
+}))
+
+const AuthForm: React.FC = () => {
+  const [showPassword, setShowPassword] = React.useState<boolean>()
+  const [errorEmailMessage, setErrorEmailMessage] = useState<boolean>(false)
+  const [errorPasswordMessage, setErrorPasswordMessage] =
+    useState<boolean>(false)
+  const styles = useStyles()
+  const authCtx = useContext(AuthContext)
+  const { enteredEmail } = authCtx
+  const { enteredPassword } = authCtx
+  const { enterEmail } = authCtx
+  const { enterPassword } = authCtx
+  const { isLogin } = authCtx
+  const { handleIsLogin } = authCtx
+  const { fetchData } = authCtx
+  const { isLoading } = authCtx
+  const history = useHistory()
+
+  const handleClickShowPassword = () => {
+    setShowPassword(!showPassword)
+  }
+
+  const handleMouseDownPassword = (event: React.MouseEvent) => {
+    event.preventDefault()
+  }
+
+  const switchAuthModeHandler = () => {
+    handleIsLogin()
+  }
+  const passwordChangeHandler = (e: React.ChangeEvent<HTMLInputElement>) => {
+    enterPassword(e.target.value)
+  }
+
+  const emailChangeHandler = (e: React.ChangeEvent<HTMLInputElement>) => {
```

Comment:
check this way to make syntax cleaner https://stackoverflow.com/questions/67143951/typescript-and-react-what-is-the-proper-type-for-an-event-handler-that-uses-des

Reply:
I`ve made some updates, but still not sure I got the point))